src/exchanges/ssr.ts

```diff
@@ -0,0 +1,127 @@
+import { pipe, share, filter, merge, map, tap } from 'wonka';
+import { Exchange, OperationResult, Operation } from '../types';
+import { CombinedError } from '../utils';
+
+export interface SerializedResult {
+  data?: any;
+  error?: {
+    networkError?: string;
+    graphQLErrors: string[];
+  };
+}
+
+export interface SSRData {
+  [key: string]: SerializedResult;
+}
+
+export interface SSRExchangeParams {
+  initialState?: SSRData;
+}
+
+export interface SSRExchange extends Exchange {
+  /** Rehydrates cached data */
+  restoreData(data: SSRData): void;
+  /** Extracts cached data */
+  extractData(): SSRData;
+}
+
+const shouldSkip = ({ operationName }: Operation) =>
+  operationName !== 'subscription' && operationName !== 'query';
+
+/** Serialize an OperationResult to plain JSON */
+const serializeResult = ({
+  data,
+  error,
+}: OperationResult): SerializedResult => {
+  const result: SerializedResult = { data, error: undefined };
+  if (error !== undefined) {
+    result.error = {
+      networkError: '' + error.networkError,
+      graphQLErrors: error.graphQLErrors.map(x => '' + x),
+    };
+  }
+
+  return result;
+};
+
+/** Deserialize plain JSON to an OperationResult */
+const deserializeResult = (
+  operation: Operation,
+  result: SerializedResult
+): OperationResult => {
+  const { error, data } = result;
+  const deserialized: OperationResult = { operation, data, error: undefined };
+  if (error !== undefined) {
+    deserialized.error = new CombinedError({
+      networkError: new Error(error.networkError),
```

Comment:
Will the user find it surprising that the networkError is an `Error` instance but the graphQLErrors are strings?

Apollo seems to wrap these into `GraphQLError`s

Reply:
We do have the graphQLErrors array as well below in the a CombinedError and networkError is just for miscellaneous runtime errors.

I haven't implemented extensive error rehydration yet because I'm sure it won't come up as an issue too soon 😅 but it's definitely something that needs to be added in the future

Reply:
Ah yeah, this checks out 👍

I didn't expect the CombinedError constructor to rehydrate the GraphQL error types. gg